homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +52,148 @@ async def async_setup_entry(
 
     if coordinator.data:
         if coordinator.data.electricity:
-            entities.append(OVOEnergyLastElectricityReading(coordinator, client))
-            entities.append(
-                OVOEnergyLastElectricityCost(
+            entities = [
+                *entities,
+                OVOEnergySensor(
+                    coordinator,
+                    OVOEnergySensorEntityDescription(
+                        key=f"{DOMAIN}_{client.account_id}_last_electricity_reading",
+                        name="OVO Last Electricity Reading",
+                        device_class=DEVICE_CLASS_ENERGY,
+                        state_class=STATE_CLASS_TOTAL_INCREASING,
+                        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+                        value=lambda usage: usage.electricity[-1].consumption,
+                    ),
+                    client,
+                ),
+                OVOEnergySensor(
+                    coordinator,
+                    OVOEnergySensorEntityDescription(
+                        key=f"{DOMAIN}_{client.account_id}_last_electricity_cost",
```

Comment:
`client` is only defined in `async_setup_entry`, thus it can't be part of the key. Instead add an additional argument to `OVOEnergySensor` to pass it along. Something like `name_prefix`.
(Here set to `f"{DOMAIN}_{client.account_id}`)